src/lib/country-data.js

```diff
@@ -1,22 +1,1111 @@
+    },
+    {
+        name: 'Trinidad and Tobago',
+        code: 'tt'
+    },
+    {
+        name: 'Tunisia',
+        code: 'tn'
+    },
+    {
+        name: 'Turkey',
+        code: 'tr'
+    },
+    {
+        name: 'Turkmenistan',
+        code: 'tm'
+    },
+    {
+        name: 'Turks and Caicos Islands',
+        code: 'tc'
+    },
+    {
+        name: 'Tuvalu',
+        code: 'tv'
+    },
+    {
+        name: 'Uganda',
+        code: 'ug'
+    },
+    {
+        name: 'Ukraine',
+        code: 'ua'
+    },
+    {
+        name: 'United Arab Emirates',
+        code: 'ae'
+    },
+    {
+        name: 'United Kingdom',
+        code: 'gb'
+    },
+    {
+        name: 'United States',
+        display: 'United States of America',
+        code: 'us'
+    },
+    {
+        name: 'United States Minor Outlying Islands',
+        code: 'um'
+    },
+    {
+        name: 'Uruguay',
+        code: 'uy'
+    },
+    {
+        name: 'Uzbekistan',
+        code: 'uz'
+    },
+    {
+        name: 'Vanuatu',
+        code: 'vu'
+    },
+    {
+        name: 'Vatican City',
+        code: 'va'
+    },
+    {
+        name: 'Venezuela',
+        code: 've'
+    },
+    {
+        name: 'Vietnam',
+        code: 'vn'
+    },
+    {
+        name: 'Virgin Islands, British',
+        code: 'vg'
+    },
+    {
+        name: 'Virgin Islands, U.S.',
+        code: 'vi'
+    },
+    {
+        name: 'Wallis and Futuna',
+        code: 'wf'
+    },
+    {
+        name: 'Western Sahara',
+        code: 'eh'
+    },
+    {
+        name: 'Yemen',
+        code: 'ye'
+    },
+    {
+        name: 'Zambia',
+        code: 'zm'
+    },
+    {
+        name: 'Zimbabwe',
+        code: 'zw'
+    }
+];
+
+const countryOptions = module.exports.countryOptions = (startingCountryInfo, valueField) => (
+    startingCountryInfo.map(thisCountryInfo => (
+        {
+            value: thisCountryInfo[valueField],
+            label: thisCountryInfo.display ? thisCountryInfo.display : thisCountryInfo.name
         }
-        return 1;
+    ))
+);
+
+module.exports.lookupCountryInfo = countryCode => (
+    countryInfo.find(country => country.code === countryCode)
+);
+
+/**
+ * Function dupeCommonCountries():
+ * takes startingCountryInfo, and duplicates any number of its country labels
+ * at the beginning of the array.
+ * E.g., to have "Germany" and "Italy" at the top of the list, use:
+ *     dupeCommonCountries(['de', 'it')
+ * @param {array} startingCountryInfo array of country objects
+ *     like [{value: 'ca', label: 'Canada'}, ...]
+ * @param {array} commonCountryCodes array of country codes we should duplicate
+ * @returns {array} revised array of country objects, with duplicates inserted at beginning
+ */
+const dupeCommonCountries = module.exports.dupeCommonCountries = (startingCountryInfo, commonCountryCodes) => {
+    const newCountryInfo = startingCountryInfo.map(item => ({...item}));
+    commonCountryCodes.reverse().forEach(commonCountryCode => {
```

Comment:
Why does this get reversed?

Reply:
it's just the way i'm getting the resulting array to have the same order as the `commonCountryCodes` -- I reverse the list, then prepend the big array with each "common" country entry, one at a time.

But your question is making me think of a more straightforward way of handling this, like:

```
    const commonCountriesInfo = commonCountryCodes.map(commonCountryCode => (
        startingCountryInfo.find(country => country.code === commonCountryCode.toLowerCase())
    ));
    return [...commonCountriesInfo, ...startingCountryInfo];
```
...simpler, right?